src/reactapp/src/components/paymentMethod/components/PaymentMethodFormManager.jsx

```diff
@@ -7,21 +7,27 @@ import { __ } from '../../../i18n';
 import { PAYMENT_METHOD_FORM } from '../../../config';
 import useFormSection from '../../../hook/useFormSection';
 import { formikDataShape } from '../../../utils/propTypes';
+import useCheckoutFormContext from '../../../hook/useCheckoutFormContext';
 import PaymentMethodFormContext from '../context/PaymentMethodFormContext';
 import usePaymentMethodAppContext from '../hooks/usePaymentMethodAppContext';
 import usePaymentMethodCartContext from '../hooks/usePaymentMethodCartContext';
 
-const initialValues = {
+const defaultValues = {
   code: '',
 };
 
 const requiredMessage = __('Required');
 
-const validationSchema = {
+const initialValidationSchema = {
   code: YupString().required(requiredMessage),
 };
 
 function PaymentMethodFormManager({ children, formikData }) {
+  const [initialValues, setInitialValues] = useState(defaultValues);
+  const [validationSchema, setValidationSchema] = useState(
```

Comment:
Make `validationSchema` states will be beneficial especially when the payment integrations needs to update the validation schema of payment section altogether.